### ISSUE TITLE
Separate Core MAUI platform and Native Controls

### ISSUE THREAD

**Issue by Martinn2** [dotnet/maui#4721]
### Description

MAUI platform SDKs should be separated from Native Controls.
It would be used like this:
```csharp
builder.UseMauiCore()
builder.UseMauiNativeControls()
```
This way, MAUI might support Linux (and maybe web) just by adding "Maui Core" for now, so support would be easier and faster.
To create UI, we would then use Blazor,Microsoft.Maui.Graphics.Controls, or any similar library (for example Syncfusion has a lot of drawn controls).

Also developers who would like to use drawn controls on other platforms would make their projects smaller maybe by removing XAML at all.

### Public API Changes

```csharp

builder.UseMauiCore()
builder.UseMauiNativeControls()
```

### Intended Use-Case

I want to be able to use MAUI without native controls to make it more lightwidth and easier to add new platforms.

**Comment by jfversluis** [comment]
How is this not the case already? For instance [Comet](https://github.com/dotnet/Comet) is already using core without using the "XAML controls" so to speak, as does [Microsoft.Maui.Graphics.Controls](https://github.com/dotnet/Microsoft.Maui.Graphics.Controls). What are you missing right now that prevents you from doing this?

**Comment by PrestigeDevop** [comment]
[Meet WebWindow, a cross-platform webview library for .NET Core](https://blog.stevensanderson.com/2019/11/18/2019-11-18-webwindow-a-cross-platform-webview-for-dotnet-core/)

It's like Electron, but without bundling Node.js or Chromium, and without most of the the APIs

**Comment by Martinn2** [comment]
@PrestigeDevop Yes, this is similar concept. If MAUI is split the way I suggest it would be easy build your library above MAUI and linux support for MAUI would be done. IMHO we need drawn controls not only Blazor, as your library has

**Comment by osnipezzini** [comment]
So come on, instead of having a complete SDK to write so much for Linux, android, windows, ios... , do you want us to write linux screens by hand?? so if so use pure GTK#.

  What's the need to have a half SDK? I do want to write linux screens using XAML, otherwise you don't even need to add linux if you need to do it separately. Today I use pure GTK#, but it lacks some things like support for generic types and lists, ease of creating tables, among other things.

MAUI with full linux support would be handy for building these applications.

**Comment by Martinn2** [comment]
@osnipezzini  With the architecture I suggest we would be able to use all the newest .NET features (like generics you mentioned) and paint with skia or native drawing library or blazor. 

```do you want us to write linux screens by hand??  so if so use pure GTK#.```
Yes, I want to write screens by hand but NOT using GTK#, but using much easier and cross platform way
